packages/server/src/tests/header/headerSelection.test.ts

```diff
@@ -0,0 +1,221 @@
+        expect(result_1_variant).toHaveProperty('name');
+        expect(result_1_variant.name).toBe('remote');
+
+        // Mock targeting data: is a supporter, not in UK
+        const mockTargetingObject_2: HeaderTargeting = {
+            showSupportMessaging: false,
+            edition: 'UK',
+            countryCode: 'ck',
+            modulesVersion: 'v3',
+            mvtId: 900263,
+        };
+
+        const result_2: HeaderTestSelection | null = selectBestTest(
+            mockTargetingObject_2,
+            mockTests,
+        );
+        const result_2_test: HeaderTest | NullReturn = result_2
+            ? result_2.test
+            : testHasReturnedNull;
+        const result_2_variant: HeaderVariant | NullReturn = result_2
+            ? result_2.variant
+            : variantHasReturnedNull;
+        expect(result_2).toBeDefined();
+        expect(result_2).toHaveProperty('test');
+        expect(result_2).toHaveProperty('variant');
+        expect(result_2).toHaveProperty('moduleName');
+        expect(result_2).toHaveProperty('modulePathBuilder');
+        expect(result_2_test).toHaveProperty('name');
+        expect(result_2_test.name).toBe('header-supporter');
+        expect(result_2_variant).toHaveProperty('name');
+        expect(result_2_variant.name).toBe('control');
+
+        // Mock targeting data: not a supporter, is in UK
+        const mockTargetingObject_3: HeaderTargeting = {
+            showSupportMessaging: true,
+            edition: 'UK',
+            countryCode: 'im', // Isle of Man (UK sterling region)
+            modulesVersion: 'v3',
+            mvtId: 900263,
+        };
+
+        const result_3: HeaderTestSelection | null = selectBestTest(
+            mockTargetingObject_3,
+            mockTests,
+        );
+        const result_3_test: HeaderTest | NullReturn = result_3
+            ? result_3.test
+            : testHasReturnedNull;
+        const result_3_variant: HeaderVariant | NullReturn = result_3
+            ? result_3.variant
+            : variantHasReturnedNull;
+        expect(result_3).toBeDefined();
+        expect(result_3).toHaveProperty('test');
+        expect(result_3).toHaveProperty('variant');
+        expect(result_3).toHaveProperty('moduleName');
+        expect(result_3).toHaveProperty('modulePathBuilder');
+        expect(result_3_test).toHaveProperty('name');
+        expect(result_3_test.name).toBe('RemoteRrHeaderLinksTest__UK');
+        expect(result_3_variant).toHaveProperty('name');
+        expect(result_3_variant.name).toBe('remote');
+
+        // Mock targeting data: is a supporter, is in UK
+        const mockTargetingObject_4: HeaderTargeting = {
+            showSupportMessaging: false,
+            edition: 'UK',
+            countryCode: 'im',
+            modulesVersion: 'v3',
+            mvtId: 900263,
+        };
+
+        const result_4: HeaderTestSelection | null = selectBestTest(
```

Comment:
not a big deal, but this file would be more maintainable if you split it into separate test cases as we do elsewhere, e.g. https://github.com/guardian/support-dotcom-components/blob/main/packages/server/src/lib/dates.test.ts

E.g
```
it('returns a UK non-supporter header', () => {
  const targeting: HeaderTargeting = {...}
  const result = selectBestTest(targeting, tests)
  expect...
}

it('returns a non-UK supporter header', () => {
  const targeting: HeaderTargeting = {...}
  const result = selectBestTest(targeting, tests)
  expect...
}
```

Reply:
Agreed, and done